### PR TITLE
hook switches user to Mainnet if invalid chain is connected

### PR DESCRIPTION
**Pull request by personnamedmike**
Linked to issue #46 .
The UX has room for improvement, but if the user isn't connected to chain 1 or 5, MetaMask will prompt the user to switch to mainnet.

https://github.com/airswap/airswap-voter-rewards/assets/90465420/43d871c4-bf68-4b09-8c76-b08628cf50f8

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "@react-hookz/web": "^23.1.0",
     "@tanstack/react-query": "^4.32.0",
     "bignumber.js": "^9.1.1",
+    "ethers": "^6.7.1",
```
Comment (gpxl-dev) [inline package.json:18]: adding ethers when we have viem already is definitely adding a lot of redundant code.

Check [this out](https://wagmi.sh/core/actions/switchNetwork)

File: src/features/staking/StakingModal.tsx [hunk 2]
```diff
@@ -2,7 +2,6 @@ import { FC, RefObject, useState } from "react";
 import { useForm } from "react-hook-form";
 import { ImSpinner8 } from "react-icons/im";
 import { VscChromeClose } from "react-icons/vsc";
-import { useForm } from "react-hook-form";
```
Comment (gpxl-dev) [inline src/features/staking/StakingModal.tsx:5]: If these are small changes that don't warrant a branch that's totally fine, just a note to check that it isn't an error.

File: package.json (hunk 1 above)
Comment (gpxl-dev) [reply package.json:18]: I don't think you are even using it 😅

File: src/hooks/useConnectToValidChain.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { useNetwork } from "wagmi";
+
+/**
+ * @remarks if user is not connected to supported chains (Mainnet 1, or Goerli 5), this hook will prompt user to switch to Mainnet
+ * @returns void
+ */
+export const useConnectToValidChain = async () => {
```
Comment (gpxl-dev) [inline src/hooks/useConnectToValidChain.tsx:7]: This function _works_, but is not quite the "react way". 

Switching chains is a side effect, which should hint that it should be in a `useEffect`. Basically anything with an async request should be.

The component that uses this hook could re-render many times in a short space of time (especially on an initial load with multiple data fetches that each set a state). In the event that the switch chain request was slower to resolve than the rerenders, it'd be firing off a lot of requests.

TLDR, wrap everything from line 10 onwards in a `useEffect`.

File: src/hooks/useConnectToValidChain.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  const { chain } = useNetwork();
+
+  if (chain?.id === 1 || chain?.id === 5) {
+    return;
+  }
+
+  try {
+    await window.ethereum.request({
```
Comment (gpxl-dev) [inline src/hooks/useConnectToValidChain.tsx:15]: may not work for all wallets - better to use the hook mentioned at the start of this review.

**Review by gpxl-dev — CHANGES_REQUESTED** [review]

File: src/features/staking/StakingModal.tsx (hunk 2 above)
Comment (personnamedmike) [reply src/features/staking/StakingModal.tsx:5]: Actually This was a change that was made in the "small bug fixes" branch. I made the same change on this branch because it was breaking the app. This change won't happen again in this PR

File: src/hooks/useConnectToValidChain.tsx (hunk 3 above)
Comment (personnamedmike) [reply src/hooks/useConnectToValidChain.tsx]: I switched everything with the wagmi hook `useSwitchNetwork` and removed the other sloppy errors. Now it's more the React way

File: src/features/staking/utils/helpers.ts [hunk 5]
```diff
@@ -112,7 +112,7 @@ export const handleButtonActions = ({
   approve: (() => Promise<WriteContractResult>) | undefined;
   stake: (() => void) | undefined;
   unstake: (() => void) | undefined;
-  setValue: UseFormSetValue<{ stakingAmount: number }>;
```
Comment (personnamedmike) [inline src/features/staking/utils/helpers.ts:115]: Same as mentioned above - these are small changes that were made in the "merge fixes" branch, so these won't get merged again into `main`

**Review by gpxl-dev — APPROVED** [review]
